lib/modules.nix

```diff
@@ -1424,20 +1424,118 @@ let
   # as a definition, as the new definition will not keep the mkOverride /
   # mkDefault properties of the previous option.
   #
-  mkAliasDefinitions = mkAliasAndWrapDefinitions id;
-  mkAliasAndWrapDefinitions = wrap: option: mkAliasIfDef option (wrap (mkMerge option.definitions));
+  mkAliasDefinitions = mkAliasDefinitionsWith {
+    withPriority = false;
+    # Set `withLocation = false` to preserve existing behaviour. TODO: Should it be true?
+    withLocation = false;
+  };
+
+  mkAliasAndWrapDefinitions =
+    wrapper:
+    mkAliasDefinitionsWith {
+      inherit wrapper;
+      withPriority = false;
+      # Set `withLocation = false` to preserve existing behaviour. TODO: Should it be true?
+      withLocation = false;
+    };
 
   # Similar to mkAliasAndWrapDefinitions but copies over the priority from the
   # option as well.
   #
   # If a priority is not set, it assumes a priority of defaultOverridePriority.
   mkAliasAndWrapDefsWithPriority =
-    wrap: option:
+    wrapper:
+    mkAliasDefinitionsWith {
+      inherit wrapper;
+      # Set `withLocation = false` to preserve existing behaviour. TODO: Should it be true?
+      withLocation = false;
```

Comment:
thought: We should probably `s/loc/prefix` in many places to remove that ambiguity, but this is the rightful meaning of the word, so no action required here.